google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITObjectChecksumSupportTest.java

```diff
@@ -0,0 +1,214 @@
+import com.google.cloud.storage.Storage.BlobWriteOption;
+import com.google.cloud.storage.StorageException;
+import com.google.cloud.storage.StorageFixture;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteStreams;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public final class ITObjectChecksumSupportTest {
+
+  @ClassRule(order = 1)
+  public static final StorageFixture storageFixtureHttp = StorageFixture.defaultHttp();
+
+  @ClassRule(order = 1)
+  public static final StorageFixture storageFixtureGrpc = StorageFixture.defaultGrpc();
+
+  @ClassRule(order = 2)
+  public static final BucketFixture bucketFixtureHttp =
+      BucketFixture.newBuilder()
+          .setBucketNameFmtString("java-storage-http-%s")
+          .setHandle(storageFixtureHttp::getInstance)
+          .build();
+
+  @ClassRule(order = 2)
+  public static final BucketFixture bucketFixtureGrpc =
+      BucketFixture.newBuilder()
+          .setBucketNameFmtString("java-storage-grpc-%s")
+          .setHandle(storageFixtureHttp::getInstance)
+          .build();
+
+  @Rule public final TestName testName = new TestName();
+
+  private final String clientName;
+  private final Storage storage;
+  private final BucketFixture bucketFixture;
+  private final ChecksummedTestContent content;
+
+  public ITObjectChecksumSupportTest(
+      String clientName,
+      StorageFixture storageFixture,
+      BucketFixture bucketFixture,
+      ChecksummedTestContent content) {
+    this.clientName = clientName;
+    this.storage = storageFixture.getInstance();
+    this.bucketFixture = bucketFixture;
+    this.content = content;
+  }
+
+  @Parameters(name = "{0}")
+  public static Iterable<Object[]> parameters() {
+    ImmutableList<Object[]> clientsAndBuckets =
+        ImmutableList.of(
+            new Object[] {"JSON/Prod", storageFixtureHttp, bucketFixtureHttp},
+            new Object[] {"GRPC/Prod", storageFixtureGrpc, bucketFixtureGrpc});
+
+    DataGenerator gen = DataGenerator.base64Characters();
+    int _2MiB = 2 * 1024 * 1024;
+    int _24MiB = 24 * 1024 * 1024;
+    ImmutableList<ChecksummedTestContent> contents =
+        ImmutableList.of(
+            // small, single message single stream when resumable
+            ChecksummedTestContent.of(gen.genBytes(15)),
+            // med, multiple messages single stream when resumable
+            ChecksummedTestContent.of(gen.genBytes(_2MiB + 3)),
+            // large, multiple messages and multiple streams when resumable
+            ChecksummedTestContent.of(gen.genBytes(_24MiB + 5)));
```

Comment:
why the 3 and 5 magic numbers being added?

Reply:
just some number to push it past the 2MiB evenly divisible values